admob/src/ios/rewarded_ad_internal_ios.mm

```diff
@@ -30,23 +30,148 @@
     ad_load_callback_data_(nil), rewarded_ad_(nil),
     parent_view_(nil), rewarded_ad_delegate_(nil) {}
 
-RewardedAdInternalIOS::~RewardedAdInternalIOS() { }
+RewardedAdInternalIOS::~RewardedAdInternalIOS() {
+  firebase::MutexLock lock(mutex_);
+  // Clean up any resources created in RewardedAdInternalIOS.
+  Mutex mutex(Mutex::kModeNonRecursive);
+  __block Mutex *mutex_in_block = &mutex;
+  mutex.Acquire();
+  void (^destroyBlock)() = ^{
+    ((GADRewardedAd*)rewarded_ad_).fullScreenContentDelegate = nil;
+    rewarded_ad_delegate_ = nil;
+    rewarded_ad_ = nil;
+    if(ad_load_callback_data_ != nil) {
+      delete ad_load_callback_data_;
+      ad_load_callback_data_ = nil;
+    }
+    mutex_in_block->Release();
+  };
+  util::DispatchAsyncSafeMainQueue(destroyBlock);
```

Comment:
Since this is a pattern we've done before, I wonder if we should actually make a DispatchAsyncSafeMainQueueBlocking, and also add tests for it.

Reply:
I'm going to change this in the next PR which includes some other stability fixes.